WalletCore/Classes/ViewModel/AssetsFilterViewModel.swift

```diff
@@ -24,11 +24,23 @@ public class AssetsFilterViewModel {
     /// On each filter event filteredAssets will be updated accordingly
     public let filter = Variable<FilterType>(.all)
     
+    public let filteredSum = Variable<Decimal>(0.0)
+    
     public init(assetsToFilter: Observable<[Variable<Asset>]>) {
         filteredAssets = Observable
             .combineLatest(assetsToFilter, filter.asObservable())
             .filterByType()
             .asDriver(onErrorJustReturn: [])
+        
+        filteredAssets.asObservable()
+            .map({ assets in
```

Comment:
You can reduce this whole `.map` method with this:
`.map { $0.map { $0.value.realCurrency.value }.reduce(0.0, +) }`
Also, you should not bind the mapped value to the Variable object, as it is not disposed. Instead assign it to 
`public let filteredSum: Observable<Decimal>`